manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,357 @@
+
+   private final class ConnectorImplementation implements Connector {
+      private final String[] schemaTable;
+
+      private ConnectorImplementation(String[] schemaTable) {
+         this.schemaTable = schemaTable;
+      }
+
+      @Override
+      public String schema() {
+         return schemaTable[0];
+      }
+
+      @Override
+      public String table() {
+         return schemaTable[1];
+      }
+   }
+
+   final EagerCacheRuleSpec eagerRule;
+
+   public EagerCacheRuleSpecAdapter(EagerCacheRuleSpec eagerRule) {
+      this.eagerRule = eagerRule;
+   }
+
+   @Override
+   public Connector connector() {
+      String[] schemaTable = eagerRule.getTableName().split("\\.");
+      return switch (schemaTable.length) {
+         case 1 -> new Connector() {
+            @Override
+            public String schema() {
+               return "";
+            }
+
+            @Override
+            public String table() {
+               return schemaTable[0];
+            }
+         };
+         case 2 -> new ConnectorImplementation(schemaTable);
+         default ->
+            throw new IllegalArgumentException("Unsupported schema format (must be table or schema.table)");
+      };
+   }
+
+   @Override
+   public KeyType keyType() {
+      switch (eagerRule.getKey().getFormat()) {
+         case TEXT:
+            return KeyType.TEXT;
+         case JSON:
+            return KeyType.JSON;
+         case UNRECOGNIZED:
+            throw new IllegalArgumentException("Unsupported KeyType");
+      }
+      throw new IllegalArgumentException("Unsupported KeyType");
+   }
+
+   @Override
+   public String plainSeparator() {
+      return eagerRule.getKey().getKeySeparator();
+   }
+
+   @Override
+   public String selectStatement() {
+      var valueSelect = (eagerRule.getValue() != null && eagerRule.getValue().getValueColumnsCount() > 0)
+            ? String.join(",", eagerRule.getValue().getValueColumnsList())
+            : "*";
+      return "select " + valueSelect + " from " + eagerRule.getTableName();
+   }
+
+   @Override
+   public boolean queryEnabled() {
+      return eagerRule.getQuery().getEnabled();
+   }
+}
+
+// Used only by CacheRuleInformer
+class LazyCacheRuleSpecAdapter implements Rule {
+
+   private final class ConnectorImplementation implements Connector {
+      private final String[] schemaTable;
+
+      private ConnectorImplementation(String[] schemaTable) {
+         this.schemaTable = schemaTable;
+      }
+
+      @Override
+      public String schema() {
+         return schemaTable[0];
+      }
+
+      @Override
+      public String table() {
+         return schemaTable[1];
+      }
+   }
+
+   final LazyCacheRuleSpec lazyRule;
+
+   public LazyCacheRuleSpecAdapter(LazyCacheRuleSpec LazyRule) {
+      this.lazyRule = LazyRule;
+   }
+
+   @Override
+   public Connector connector() {
+      // TODO: how to implement this for lazy rules?
+      // We don't have a table field in the k8s spec
```

Comment:
This is only used by the `DatabaseHandler` which is not required for Lazy rules.

I think we have two options here:

1. Proceed with a single Rule configuration for both Eager & Lazy rules, in which case we need to add a field to differentiate at runtime between the two.
2. Create separate EagerRule and LazyRule configuration. The DatabaseHandler can simply process the EagerRule configs.

My preference is for 2, as I think the Eazy and Lazy rules will only continue to diverge as we add more features. @jabolina @pruivo @wburns WDYT?

Reply:
I also guess 2 is better

Reply:
+1 to number 2

Reply:
Definitely 2 from me.